### PR TITLE
net-im/mattermost-desktop-bin: version bump to 3.7.0

### PR DESCRIPTION
**Pull request by mhz-tamb**

### CODE REVIEW

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+inherit eutils gnome2-utils
+
+MY_PN="${PN%-*}"
+
+DESCRIPTION="Mattermost Desktop application"
+HOMEPAGE="https://about.mattermost.com/"
+
+SRC_URI="
+https://github.com/mattermost/desktop/archive/v${PV}.tar.gz -> ${P}.tar.gz
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: Misindent?

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	doins locales/*.pak
+
+	insinto "/opt/${MY_PN}/resources"
+	doins resources/*.asar
+
+	insinto "/opt/${MY_PN}"
+	doins *.pak *.bin *.dat
+
+	exeinto "/opt/${MY_PN}"
+
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: Stray line?

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	doexe *.so "${MY_PN}"
+
+	dosym "/opt/${MY_PN}/${MY_PN}" "/usr/bin/${MY_PN}"
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: Please do not use absolute symlinks. Make it a relative symlink instead (use relative path in first arg).

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+	newicon "${S}/icon.png" "${MY_PN}.png"
+	make_desktop_entry "${MY_PN}" Mattermost "${MY_PN}"
+}
+
+pkg_preinst() {
+	gnome2_icon_savelist
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: This is no longer needed these days.

**Comment by mhz-tamb** [comment]
thank you! it's okay now?

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+DESCRIPTION="Mattermost Desktop application"
+HOMEPAGE="https://about.mattermost.com/"
+
+SRC_URI="
+	https://github.com/mattermost/desktop/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	amd64? ( https://releases.mattermost.com/desktop/${PV}/mattermost-desktop-${PV}-linux-x64.tar.gz )
+	x86?   ( https://releases.mattermost.com/desktop/${PV}/mattermost-desktop-${PV}-linux-ia32.tar.gz )
+"
+
+LICENSE="Apache-2.0"
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: AFAICS this bundles a number of libraries (libffmpeg, libnode…). You need to include licenses for them as well.

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RESTRICT="mirror"
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: Is there a specific reason for the mirroring restriction? I don't think the license requires that.

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+DESCRIPTION="Mattermost Desktop application"
+HOMEPAGE="https://about.mattermost.com/"
+
+SRC_URI="
+	https://github.com/mattermost/desktop/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	amd64? ( https://releases.mattermost.com/desktop/${PV}/mattermost-desktop-${PV}-linux-x64.tar.gz )
+	x86?   ( https://releases.mattermost.com/desktop/${PV}/mattermost-desktop-${PV}-linux-ia32.tar.gz )
+"
+
+LICENSE="Apache-2.0 GPL v2+ LGPL-2.1+ MIT"
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: Those LICENSE thingies have to correspond to files from /licenses directory.

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+	insinto "/opt/${MY_PN}/resources"
+	doins resources/*.asar
+
+	insinto "/opt/${MY_PN}"
+	doins *.pak *.bin *.dat
+	exeinto "/opt/${MY_PN}"
+	doexe *.so "${MY_PN}"
+
+	dosym "../../opt/${MY_PN}/${MY_PN}" "/usr/bin/${MY_PN}"
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: I'm sorry for noticing this earlier but is there any reason it's not installed in /opt/bin? That's where we usually create symlinks for apps in /opt.

File: net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild [hunk 9]
```diff
@@ -40,7 +42,7 @@ src_install() {
 	exeinto "/opt/${MY_PN}"
 	doexe *.so "${MY_PN}"
 
-	dosym "../../opt/${MY_PN}/${MY_PN}" "/usr/bin/${MY_PN}"
+	dosym "../../opt/${MY_PN}/${MY_PN}" "/opt/bin/${MY_PN}"
```
Comment (mgorny) [inline net-im/mattermost-desktop-bin/mattermost-desktop-bin-3.7.0.ebuild]: Now you can shorten the relative path.

**Pull request closed without merging**